File: aoc2017/day1.swift
//
//  day1.swift
//  aoc2017
//
//  Created by Chris McElroy on 7/22/21.
//

import Foundation

func day1() {
	let string = input[0]
	
	var sum = 0
	for (i,c) in string.enumerated() where c == string[(i + string.count/2) % string.count] {
		sum += Int(String(c))!
	}
	
	print(sum)
}

File: go/day1.go
package main

func day1() {
	input := input[0]
	doubled := input + input
	sum := 0

	for i, c := range input {
		if input[i] == doubled[i+len(input)/2] {
			sum += int(c - '0')
		}
	}

	println(sum)
}
